src/algebra/graded_monoid.lean

```diff
@@ -394,3 +395,28 @@ instance set_like.gcomm_monoid {S : Type*} [set_like S R] [comm_monoid R] [add_c
   ..set_like.gmonoid A}
 
 end subobjects
+
+section homogeneous_elements
+
+variables {R S : Type*} [set_like S R]
+
+/-- An element `a : R` is said to be homogeneous if there is some `i : ι` such that `a ∈ A i`-/
+def set_like.is_homogeneous (A : ι → S) (a : R) : Prop := ∃ i, a ∈ A i
```

Comment:
This should probably appear in the "internal" section of the module docstring

Reply:
I have added it to doc string part. Thank you for all your help

Reply:
Looks good, thanks!